File: Practice/StaticRangeSum.cpp
#include <iostream>

using namespace std;

int main() {
    int N;
    int Q;
    cin >> N >> Q;
    long long preSum[N + 1];
    for (int i = 1; i <= N; i++) {
        int val;
        cin >> val;
        preSum[i] = preSum[i - 1] + val;
    }
    while (Q > 0) {
        long long a,b;
        cin >> a >> b;
        cout<<(preSum[b] - preSum[a - 1]) << endl;
        Q--;
    }
}

File: Practice/StaticRangeSum.java
package Practice;

import java.util.*;
import java.io.*;


class  FastScanner {
    BufferedReader br;
    StringTokenizer st;

    public FastScanner(String s) {
        try {
            br = new BufferedReader(new FileReader(s));
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
    }

    public FastScanner() {
        br = new BufferedReader(new InputStreamReader(System.in));
    }

    String nextToken() {
        while (st == null || !st.hasMoreElements()) {
            try {
                st = new StringTokenizer(br.readLine());
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return st.nextToken();
    }

    int nextInt() {
        return Integer.parseInt(nextToken());
    }

    long nextLong() {
        return Long.parseLong(nextToken());
    }

    double nextDouble() {
        return Double.parseDouble(nextToken());
    }
}

public class StaticRangeSum {
    public static void main(String[] args) {
        FastScanner sc = new FastScanner();
        PrintWriter out = new PrintWriter(System.out);

        int N = sc.nextInt();
        int Q = sc.nextInt();
        int[] preSum = new int[N + 1];
        for (int i = 1; i <= N; i++) {
            preSum[i] = preSum[i - 1] + sc.nextInt();
        }
        while (Q > 0) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            out.println(preSum[b] - preSum[a - 1]);
            Q--;
        }
        out.flush();
    }
}
